horovod/common/exceptions.py

```diff
@@ -29,3 +29,12 @@ class HostsUpdatedInterrupt(RuntimeError):
     In elastic mode, this will result in a reset event without a restore to committed state.
     """
     pass
+
+class HorovodVersionMismatchError(RuntimeError):
+    """Internal error raised when the runtime version of a framework mismatches its version at
+    Horovod installation time.
+    """
+    def __init__(self, name, version, installed_version):
+        self.name = name
```

Comment:
We should customize the error message by passing the message to the parent constructor:

```
super().__init__(message)
```